src/eos-crash-metrics.c

```diff
@@ -0,0 +1,174 @@
+  g_variant_dict_insert_value (&dict, "timestamp", g_variant_new_int16 (timestamp));
+  g_variant_dict_insert_value (&dict, "ostree_commit", g_variant_new_string (ostree_commit));
+  g_variant_dict_insert_value (&dict, "ostree_url", g_variant_new_string (ostree_url));
+
+  emtr_event_recorder_record_event_sync (emtr_event_recorder_get_default (),
+                                         PROGRAM_DUMPED_CORE_EVENT,
+                                         g_variant_dict_end (&dict));
+}
+
+static OstreeSysroot *
+load_ostree_sysroot (void)
+{
+  OstreeSysroot *sysroot = ostree_sysroot_new_default ();
+  g_autoptr(GError) error = NULL;
+
+  if (!ostree_sysroot_load (sysroot, NULL, &error))
+    {
+      g_warning ("Unable to get current OSTree sysroot: %s", error->message);
+      return NULL;
+    }
+
+  return sysroot;
+}
+
+static char *
+get_eos_ostree_repo_url (OstreeRepo *repo)
+{
+  GKeyFile *config = NULL;
+  g_autoptr(GError) error = NULL;
+  char *url = NULL;
+
+  config = ostree_repo_get_config (repo);
+
+  if (!(url = g_key_file_get_string (config, "remote \"eos\"", "url", &error)))
+    {
+      g_warning ("Unable to read OSTree config for eos remote URL: %s", error->message);
+      return NULL;
+    }
+
+  return url;
+}
+
+static char *
+get_eos_ostree_deployment_commit (OstreeSysroot *sysroot, OstreeRepo *repo)
+{
+  OstreeDeployment *deployment = ostree_sysroot_get_booted_deployment (sysroot);
+  GKeyFile *config = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree char *refspec = NULL;
+  char *commit = NULL;
+
+  if (!deployment)
+    {
+      g_warning ("OSTree deployment is not currently booted, cannot read state");
+      return NULL;
+    }
+
+  config = ostree_deployment_get_origin (deployment);
+
+  if (!(refspec = g_key_file_get_string (config, "origin", "refspec", &error)))
+    {
+      g_warning ("Unable to read OSTree refspec for booted deployment: %s", error->message);
+      return NULL;
+    }
+
+  if (!ostree_repo_resolve_rev (repo, refspec, FALSE, &commit, &error))
+    {
+      g_warning ("Unable to resolve revision for refpsec %s: %s", refspec, error->message);
+      return NULL;
+    }
+
+  return commit;
+}
+
+/* This is the way that the kernel gives us paths ... */
+static const char *blacklisted_prefixes[] =
+{
+  "!home",
+  "!sysroot!home"
+};
+
+static gboolean
+is_blacklisted_path (const char *path)
+{
+  const size_t n_blacklisted_prefixes = G_N_ELEMENTS (blacklisted_prefixes);
+  size_t i = 0;
+
+  for (; i < n_blacklisted_prefixes; ++i)
+    {
+      if (g_str_has_prefix (path, blacklisted_prefixes[i]))
+        return TRUE;
+    }
+
+  return FALSE;
+}
+
+int
+main (int argc, char **argv)
+{
+  const gchar *path = argv[1];
+  const gint16 signal = atoi (argv[2]);
+  const gint64 timestamp = atoll (argv[3]);
+  g_autoptr(OstreeSysroot) sysroot = NULL;
+  g_autoptr(OstreeRepo) repo = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree char *url = NULL;
+  g_autofree char *commit = NULL;
+
```

Comment:
I think the program should handle the case when we do not have 3 input arguments.

Reply:
In this case, I think it is fair to assume that we eithet got three arguments or were called incorrectly. I'll add a handler for that